main.py

```diff
@@ -232,6 +233,14 @@ def create_circuit(self, request: Request) -> JSONResponse:
             log.debug("create_circuit result %s %s", exception, 400)
             raise HTTPException(400, detail=str(exception)) from exception
 
+        if evc.is_intra_switch():
+            if evc.uni_a.interface.switch.status == EntityStatus.DISABLED:
```

Comment:
Shouldn't this also block if either `uni_a.interface` or `uni_z.interface` status is disabled? 

Currently, it's allowing.

Reply:
There are too many statements in this method, added `check_disabled_component()` to check for the switch and interfaces.